osu.Game/Rulesets/UI/Playfield.cs

```diff
@@ -124,6 +135,11 @@ private void load()
         /// <param name="h">The DrawableHitObject to add.</param>
         public virtual void Add(DrawableHitObject h)
         {
+            if (h.HasInitialized)
+                throw new InvalidOperationException($"{nameof(Playfield.Add)} doesn't support {nameof(DrawableHitObject)} reuse. Use pooling instead.");
```

Comment:
class name qualification is redundant

```suggestion
                throw new InvalidOperationException($"{nameof(Add)} doesn't support {nameof(DrawableHitObject)} reuse. Use pooling instead.");
```